com.civicrm.hrjobroles/js/src/job-roles/services/hr-job-roles-service.js

```diff
@@ -1,103 +1,69 @@
 define([
-    'job-roles/services/services'
-], function (services) {
+  'common/lodash',
+  'job-roles/services/services'
+], function (_, services) {
     'use strict';
 
     services.factory('HRJobRolesService', ['$log', '$q', '$filter', function ($log, $q, $filter) {
 
+        /**
+         * Extracts the contract revisions details from the chained api calls
+         * properties, then removes the current one and format the dates
+         *
+         * @param  {Object} contract
+         */
+        function processContractRevisions(contract) {
+          var contractRevisions = contract['api.HRJobContractRevision.get'].values;
+          delete(contract['api.HRJobContractRevision.get']);
+
+          contract.revisions = contractRevisions
+            .map(function (revision) {
```

Comment:
Instead of looping 3 times, we can loop once using for loop and make all the adjustments